src/metadata/metadata_hash.c

```diff
@@ -525,23 +529,28 @@ int ocf_metadata_hash_init(struct ocf_cache *cache,
 
 	if (result) {
 		ocf_metadata_hash_deinit(cache);
-	} else {
-		cache->conf_meta = METADATA_MEM_POOL(ctrl,
-				metadata_segment_sb_config);
+		return result;
+	}
 
-		/* Set core metadata */
-		cache->core_conf_meta = METADATA_MEM_POOL(ctrl,
-				metadata_segment_core_config);
+	cache->conf_meta = METADATA_MEM_POOL(ctrl,
+			metadata_segment_sb_config);
 
-		cache->core_runtime_meta = METADATA_MEM_POOL(ctrl,
-				metadata_segment_core_runtime);
+	/* Set core metadata */
+	core_meta_config = METADATA_MEM_POOL(ctrl,
+			metadata_segment_core_config);
+	core_meta_runtime = METADATA_MEM_POOL(ctrl,
+			metadata_segment_core_runtime);
 
-		env_spinlock_init(&cache->metadata.lock.eviction);
-		env_rwlock_init(&cache->metadata.lock.status);
-		env_rwsem_init(&cache->metadata.lock.collision);
+	for_each_core(cache, core, core_id) {
+		core->conf_meta = &core_meta_config[core_id];
+		core->runtime_meta = &core_meta_runtime[core_id];
 	}
 
-	return result;
+	env_spinlock_init(&cache->metadata.lock.eviction);
+	env_rwlock_init(&cache->metadata.lock.status);
+	env_rwsem_init(&cache->metadata.lock.collision);
+
+	return 0;
```

Comment:
why not just return result?

Reply:
To emphasise that when we reach this point, there is no option to fail.